src/QsCompiler/TextProcessor/QsExpressionParsing.fs

```diff
@@ -183,9 +183,22 @@ let internal numericLiteral =
             let trimmed = nl.String.TrimStart '+'
             if format = 10 || format = 0 then trimmed else trimmed.Substring 2 |> sprintf "0%s" // leading 0 is required to keep numbers positive
         let isInt    = nl.IsInteger     && format <> 0                  && nl.SuffixLength = 0 // any format is fine here
-        let isBigInt = nl.IsInteger     && (format = 10 || format = 16) && nl.SuffixLength = 1 && System.Char.ToUpperInvariant(nl.SuffixChar1) = 'L'
+        let isBigInt = nl.IsInteger     && format <> 0  && nl.SuffixLength = 1 && System.Char.ToUpperInvariant(nl.SuffixChar1) = 'L'
         let isDouble = not nl.IsInteger && format = 10                  && nl.SuffixLength = 0 
         let returnWithRange kind = preturn (kind, range)
+        let binaryToHex binaryString = 
+            // first pad 0's so that length is multiple of 4, so we can match from left rather than right
+            let nZeroPad = (4 - String.length binaryString % 4) % 4 // if str.Length is already multiple of 4 then we don't pad
+            let binaryString = binaryString.PadLeft (nZeroPad + String.length binaryString, '0')
+            // now match from left
+            binaryString |> Seq.chunkBySize 4 |> Seq.map (fun x -> System.Convert.ToByte(System.String x, 2).ToString "X") |> System.String.Concat
+        // seems like toString only takes keyword "X", so still need to enumerate for octal 
+        let octalToHex octalString =
+            // first pad 0's so that length is multiple of 4, so we can match from left rather than right
+            let nZeroPad = (4 - String.length octalString % 4) % 4 // if str.Length is already multiple of 4 then we don't pad
+            let octalString = octalString.PadLeft (nZeroPad + String.length octalString, '0')
+            // now match from left
+            octalString |> Seq.chunkBySize 4 |> Seq.map (fun x -> System.Convert.ToInt32(System.String x, 8).ToString "X") |> System.String.Concat
```

Comment:
`binaryToHex` and `octalToHex` are very similar - the only difference seems to be the parsed base (2 or 8). Can these be refactored into `let baseToHex base str = ...`, where `base` is either 2 or 8? (For simplicity we can just use `Convert.ToInt32` in both cases instead of `ToByte` for the binary one.)

Reply:
Addressed